aten/src/ATen/native/quantized/cpu/qnnpack/deps/clog/include/clog.h

```diff
@@ -0,0 +1,123 @@
+/*
```

Comment:
have we tried to recompile qnnpack code with c++ - then we can drop this and instead rely on main PyTorch's logging

Reply:
We can definitely do that but we are under somewhat of a time constraint here.  What was supposed to be a simple change (add support for runtime quantization) has already taken a long time because of all this back and forth, but I can definitely make this change if you have a strong preference.

Reply:
Let's try to make this PR a simple copy and do meaningful changes in separate PRs.
Is it a simple compiler change or do we need fix syntax / measure perf impact / etc?
Sounds like something nice to have - I feel if we have new features to be written in c++ then it's more urgent change, otherwise we can try it after finishing other critical perf work?

Reply:
Please refer to my comment above:

> OK so I tried compiling with C++, but it's a good deal of work. The compiler is complaining about jumps (use of goto to be exact) bypassing variable initializations, and use of 'restrict' (as it's not a C++ keyword.) All of this can be worked around (using __restrict for instance instead of restrict and initializing all these variables to null / zero prior at them being jumped over), but it's a good deal of work. Making these changes is actually one thing and not terribly difficult, but making sure they do not introduce any hidden bugs is a whole another can of worms that I would rather not open at this point.
> 
> https://gist.github.com/AshkanAliabadi/d7e78ce6c0ea6e9b092d71a9bb96b9de